src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -1069,22 +1067,54 @@ static List<Compilation> findDirectReferencedCompilations(Compilation compilatio
                 {
                     if (pdbStream != null)
                     {
-                        pdb = pdbStream.ToImmutable();
+                        pdb = getBytes(pdbStream);
                         pdbStream.Dispose();
                     }
                 }
 
                 diagnostics.AddRange(result.Diagnostics);
-                assembly = executableStream.ToImmutable();
+                assembly = getBytes(executableStream);
 
                 if (result.Success)
                 {
                     return new EmitOutput(assembly, pdb);
                 }
 
                 return null;
+
+                Stream getStream(string? assemblyName, string ext)
+                {
+                    if (Environment.GetEnvironmentVariable("ROSLYN_TEST_EMIT_PATH") is string { Length: > 0 } basePath)
```

Comment:
We already have a machinery to dump DLLs:

https://github.com/dotnet/roslyn/blob/4f9e5aef45e3d5255e8fcb1bf8c30020a19425af/src/Compilers/Test/Core/CompilationVerifier.cs#L40

https://github.com/dotnet/roslyn/blob/4f9e5aef45e3d5255e8fcb1bf8c30020a19425af/src/Compilers/Test/Core/CompilationVerifier.cs#L171

Should we reuse it?

I also feel like setting an env var might be difficult when running the test in VS, for example. Perhaps uncommenting an `#if` symbol would be a nice alternative.